// react-maps-google/Marker.js
"use strict";

exports.__esModule = true;
exports["default"] = void 0;

var _react = _interopRequireWildcard(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _getRequireWildcardCache(nodeInterop) { if (typeof WeakMap !== "function") return null; var cacheBabelInterop = new WeakMap(); var cacheNodeInterop = new WeakMap(); return (_getRequireWildcardCache = function _getRequireWildcardCache(nodeInterop) { return nodeInterop ? cacheNodeInterop : cacheBabelInterop; })(nodeInterop); }

function _interopRequireWildcard(obj, nodeInterop) { if (!nodeInterop && obj && obj.__esModule) { return obj; } if (obj === null || typeof obj !== "object" && typeof obj !== "function") { return { "default": obj }; } var cache = _getRequireWildcardCache(nodeInterop); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj["default"] = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

function ownKeys(object, enumerableOnly) { var keys = Object.keys(object); if (Object.getOwnPropertySymbols) { var symbols = Object.getOwnPropertySymbols(object); if (enumerableOnly) { symbols = symbols.filter(function (sym) { return Object.getOwnPropertyDescriptor(object, sym).enumerable; }); } keys.push.apply(keys, symbols); } return keys; }

function _objectSpread(target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i] != null ? arguments[i] : {}; if (i % 2) { ownKeys(Object(source), true).forEach(function (key) { _defineProperty(target, key, source[key]); }); } else if (Object.getOwnPropertyDescriptors) { Object.defineProperties(target, Object.getOwnPropertyDescriptors(source)); } else { ownKeys(Object(source)).forEach(function (key) { Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key)); }); } } return target; }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

function _inheritsLoose(subClass, superClass) { subClass.prototype = Object.create(superClass.prototype); subClass.prototype.constructor = subClass; _setPrototypeOf(subClass, superClass); }

function _setPrototypeOf(o, p) { _setPrototypeOf = Object.setPrototypeOf || function _setPrototypeOf(o, p) { o.__proto__ = p; return o; }; return _setPrototypeOf(o, p); }

var CALLBACK_MAP = {
  'animation_changed': 'onAnimationChange',
  'click': 'onClick',
  'clickable_changed': 'onClickableChange',
  'cursor_changed': 'onCursorChange',
  'dblclick': 'onDoubleClick',
  'drag': 'onDrag',
  'dragend': 'onDragEnd',
  'draggable_changed': 'onDraggableChange',
  'dragstart': 'onDragStart',
  'flat_changed': 'onFlatChange',
  'icon_changed': 'onIconChange',
  'mousedown': 'onMouseDown',
  'mouseout': 'onMouseOut',
  'mouseover': 'onMouseOver',
  'mouseup': 'onMouseUp',
  'position_changed': 'onPositionChange',
  'rightclick': 'onRightClick',
  'shape_changed': 'onShapeChange',
  'title_changed': 'onTitleChange',
  'visible_changed': 'onVisibleChange',
  'zindex_changed': 'onZIndexChange'
};

var Marker = /*#__PURE__*/function (_Component) {
  _inheritsLoose(Marker, _Component);

  function Marker() {
    return _Component.apply(this, arguments) || this;
  }

  var _proto = Marker.prototype;

  _proto.componentDidMount = function componentDidMount() {
    if (this.props.map) {
      this.renderMarker(this.props);
    }
  };

  _proto.componentDidUpdate = function componentDidUpdate(prevProps) {
    var _this$props = this.props,
        map = _this$props.map,
        position = _this$props.position,
        options = _this$props.options;
    var renderMarker = false;

    if (map !== prevProps.map) {
      renderMarker = true;
    }

    if (!Object.is(position, prevProps.position)) {
      renderMarker = true;
    }

    if (!Object.is(options, prevProps.options)) {
      renderMarker = true;
    }

    if (renderMarker) {
      this.renderMarker(this.props);
    }
  };

  _proto.componentWillUnmount = function componentWillUnmount() {
    if (!this.marker) {
      return;
    }

    google.maps.event.clearInstanceListeners(this.marker);
    this.marker.setMap(null);
  };

  _proto.onMarkerCallback = function onMarkerCallback(callback, event) {
    if (this.props[callback]) {
      this.props[callback](this.marker, event);
    }
  };

  _proto.renderMarker = function renderMarker(props) {
    var _this = this;

    var map = props.map,
        options = props.options,
        position = props.position,
        width = props.width,
        height = props.height;

    if (!map) {
      return;
    }

    if (!this.marker) {
      this.marker = new google.maps.Marker(_objectSpread(_objectSpread({}, options), {}, {
        map: map,
        position: position
      }));
      Object.keys(CALLBACK_MAP).forEach(function (key) {
        google.maps.event.addListener(_this.marker, key, _this.onMarkerCallback.bind(_this, CALLBACK_MAP[key]));
      });
      return;
    }

    this.marker.setOptions(_objectSpread(_objectSpread({}, options), {}, {
      map: map,
      position: position
    }));
  };

  _proto.render = function render() {
    return null;
  };

  return Marker;
}(_react.Component);

Marker.propTypes = {
  map: _propTypes["default"].object,
  options: _propTypes["default"].object,
  position: _propTypes["default"].object.isRequired,
  onAnimationChange: _propTypes["default"].func,
  onClick: _propTypes["default"].func,
  onClickableChange: _propTypes["default"].func,
  onCursorChange: _propTypes["default"].func,
  onDoubleClick: _propTypes["default"].func,
  onDrag: _propTypes["default"].func,
  onDragEnd: _propTypes["default"].func,
  onDraggableChange: _propTypes["default"].func,
  onDragStart: _propTypes["default"].func,
  onFlatChange: _propTypes["default"].func,
  onIconChange: _propTypes["default"].func,
  onMouseDown: _propTypes["default"].func,
  onMouseOut: _propTypes["default"].func,
  onMouseOver: _propTypes["default"].func,
  onMouseUp: _propTypes["default"].func,
  onPositionChange: _propTypes["default"].func,
  onRightClick: _propTypes["default"].func,
  onShapeChange: _propTypes["default"].func,
  onTitleChange: _propTypes["default"].func,
  onVisibleChange: _propTypes["default"].func,
  onZIndexChange: _propTypes["default"].func
};
Marker.defaultProps = {
  options: {},
  onAnimationChange: function onAnimationChange(marker) {},
  onClick: function onClick(marker, event) {},
  onClickableChange: function onClickableChange(marker) {},
  onCursorChange: function onCursorChange(marker) {},
  onDoubleClick: function onDoubleClick(marker, event) {},
  onDrag: function onDrag(marker, event) {},
  onDragEnd: function onDragEnd(marker, event) {},
  onDraggableChange: function onDraggableChange(marker) {},
  onDragStart: function onDragStart(marker, event) {},
  onFlatChange: function onFlatChange(marker) {},
  onIconChange: function onIconChange(marker) {},
  onMouseDown: function onMouseDown(marker, event) {},
  onMouseOut: function onMouseOut(marker, event) {},
  onMouseOver: function onMouseOver(marker, event) {},
  onMouseUp: function onMouseUp(marker, event) {},
  onPositionChange: function onPositionChange(marker) {},
  onRightClick: function onRightClick(marker, event) {},
  onShapeChange: function onShapeChange(marker) {},
  onTitleChange: function onTitleChange(marker) {},
  onVisibleChange: function onVisibleChange(marker) {},
  onZIndexChange: function onZIndexChange(marker) {}
};
var _default = Marker;
exports["default"] = _default;
